packages/create-vite/__tests__/cli.spec.ts

```diff
@@ -69,24 +79,92 @@ test('asks to overwrite non-empty current directory', () => {
 })
 
-test('successfully scaffolds a project based on vue starter template', () => {
-  const { stdout } = run([projectName, '--template', 'vue'], {
+test('answer n to prompt for overwrite non-empty current directory', () => {
+  createNonEmptyDir()
+  const { stdout } = run([projectName], { cwd: __dirname, input: 'n' })
+  expect(stdout).toContain('Operation cancelled')
+})
+
+test('prompt for install and start it now', () => {
+  const { stdout } = run([projectName, '-t', 'react'], {
     cwd: __dirname
   })
-  const generatedFiles = readdirSync(genPath).sort()
-
-  // Assertions
-  expect(stdout).toContain(`Scaffolding project in ${genPath}`)
-  expect(templateFiles).toEqual(generatedFiles)
+  expect(stdout).toContain(`Install and start it now?`)
 })
 
-test('works with the -t alias', () => {
+test('answer n to prompt for install and start it now', () => {
   const { stdout } = run([projectName, '-t', 'vue'], {
+    cwd: __dirname,
+    input: 'n'
+  })
+  scaffoldsProjectExpect(stdout)
+})
+
+test('prompt for choose the agent', () => {
+  const { stdout } = run([projectName, '--template', 'preact'], {
+    cwd: __dirname,
+    input: 'y'
+  })
+  expect(stdout).toContain(`Choose the agent:`)
+})
+
+test('successfully scaffolds a project with vue template and -i', () => {
+  const { stdout } = run([projectName, '--template', 'vue', '-i'], {
     cwd: __dirname
   })
-  const generatedFiles = readdirSync(genPath).sort()
+  scaffoldsProjectExpect(stdout)
+})
 
-  // Assertions
-  expect(stdout).toContain(`Scaffolding project in ${genPath}`)
-  expect(templateFiles).toEqual(generatedFiles)
+test('successfully scaffolds a project with vue template and --immediate to non-empty directory', () => {
+  createNonEmptyDir()
+  const { stdout } = run([projectName, '--template', 'vue', '--immediate'], {
+    cwd: __dirname,
+    input: 'y'
+  })
+  scaffoldsProjectExpect(stdout)
 })
+
+const speicalJestTime = 60 * 1000
+const expectRunTime = 30 * 1000
+test(
+  'successfully start a project by npm agent',
+  async () => {
+    const subprocess = runAsync([projectName, '-t', 'vanilla'], {
+      cwd: __dirname
+    })
+    const [waiting, resolveWaiting] = (() => {
+      let resolve$
+      const waiting = new Promise<void>((resolve) => {
+        resolve$ = resolve
+      })
+      const race = setTimeout(() => {
+        resolve$()
+      }, expectRunTime)
+      return [
+        waiting,
+        () => {
+          clearTimeout(race)
+          resolve$()
+        }
+      ]
+    })()
+    let text = ''
+    const { stdin, stdout } = subprocess
+    stdout.on('data', (data) => {
+      text = data.toString()
+      if (text.includes('Install and start it now?')) {
+        stdin.write('y\n')
+      }
+      if (text.includes('Choose the agent:')) {
+        stdin.write('\n')
+      }
+      if (text.includes('ready in')) {
+        resolveWaiting()
+      }
+    })
+    await waiting
+    subprocess.kill('SIGHUP')
```

Comment:
It seems that this test fails in windows...

Reply:
Anybody have an idea how to write this test?

Reply:
Try adding `process.exit(0)` to satisfy the process was successful.

Reply:
Thanks for your reply, i think using `process.exit` will close the main process. I find a possible solution, maybe we can use `taskkill` on Windows.

Reply:
yeah, you would need to detect it based on the **Operating system**

Reply:
yeah